### ISSUE TITLE
`debounced:scroll` on Scrollable Elements

### ISSUE THREAD

**Issue by andrewerlanger** [hopsoft/debounced#8]
First and foremost, many thanks for the great library – it's a real joy to work with :)

Just thought I'd make a note of one minor hiccup I think I've found when it comes to scrolling on individual elements rather than the document.

Let's say we have an HTML structure along these lines, with elements on the page that are individually scrollable:

```html
<body style="display: flex">
  <nav id="sidebar" style="overflow-y: scroll">
    ...
  </nav>
  <div id="main-pane" style="overflow-y: scroll">
    ...
  </div>
</body>
```

Here, scrolling on either the `sidebar` or the `main-pane` will fire a `scroll` event on the element in question, but this event will *not* bubble up to the `document` as far as I understand.

Because `debounced:scroll` is ultimately handled via an event listener on the `document` ([this part here](https://github.com/hopsoft/debounced/blob/914838504d9eb1627ab84057660eb29557128ee2/src/index.js#L33)), I don’t think it’s currently possible to listen for a debounced scroll on the scrollable elements themselves. This means something like the following will not be fired on scroll:

```js
const sidebar = document.getElementById("sidebar");
sidebar.addEventListener("debounced:scroll", () => console.log("scrolling..."));
```

If this is the case then I don’t have a neat solution in mind unfortunately – just thought I’d post in here in case someone else might!

**Comment by hopsoft** [comment]
This is correct given that scroll events on `Element` don't bubble, but I'll give this some more consideration for the next release.

**Comment by hopsoft** [comment]
This will be resolved by #23 

<img width="2772" height="1374" alt="Image" src="https://github.com/user-attachments/assets/896d10ff-e07c-4d62-8861-d098624e2baf" />
